### PR TITLE
Module 2 exercise

### PR DESCRIPTION
**Pull request by MadhuRoy-git**
I have called this branch TelloAPIIntegration. I had created a module_2 branch but didn't use that.Please consider this as my Module 2 branch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -114,3 +114,6 @@ env/
 
 ### Configuration files ###
 .env
+
+### Secrets files ###
+secrets.py
```
Comment (hugh-emerson) [inline .gitignore]: This should be mentioned in the README. It should contain anything a developer needs to know to get the project running after cloning the repo.

But also, it shouldn't be necessary to have an ignored python file. Use the .env file instead to set environment variables and then load them in python code with `os.getenv('YOUR_ENV_VAR_NAME')`

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,12 @@
+certifi==2020.6.20
+chardet==3.0.4
+click==7.1.2
+Flask==1.1.2
```
Comment (hugh-emerson) [inline requirements.txt:4]: You have two entries for Flask now, which breaks the file, and did you deliberately remove python-dotenv?

It's used for e.g. automatic loading of the .env file when running `flask run`.

File: templates/index.html [hunk 3]
```diff
@@ -3,19 +3,21 @@ <h1>To-Do App</h1>
...
+  <button class="btn btn-outline-primary" type="submit">Undo</button>
+</form>
 <p class="p5"><span class="s1">{% endfor %}</span></p>
 <form action="/add" method="post">
-  <label>Enter title of new item to be added</label>
-  <br/>
-  <input name="input_title" type="text" required placeholder="title" id="exampleInput1">
-  <button type="submit">Add</button>
+  <label for="exampleInput1">Item to be added:</label><br>
+  <input type="text" name="new_item_name" placeholder="title" id="exampleInput1">
```
Comment (hugh-emerson) [inline templates/index.html]: You should probably replace "exampleInput1" on these lines with a meaningful id. 

Also HTML attributes are case insensitive so it's better to use kebab-case or snake_case.

File: trello_items.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    """
+
+    query = {
+        'key': key,
+        'token': token
+    }
+   
+    board_id = secrets.boardId
+    to_do_cards = []
+    url = "https://api.trello.com/1/boards/{}/cards".format(board_id)
```
Comment (hugh-emerson) [inline trello_items.py]: If you have an up to date version of python you should be able to write `f"https://api.trello.com/1/boards/{board_id}/cards"` which is easier to read.

File: trello_items.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    cards = requests.request("GET", url, params=query)
+    for card in cards.json():
+        if (get_list_name(card['id']) == 'To Do'):
+            to_do_cards.append(Card(card['id'], card['pos'], card['name'], card['desc'], get_list_name(card['id'])))
```
Comment (hugh-emerson) [inline trello_items.py]: You already know the list name is 'To Do'. This line is inside an if which checks it.

As written, you're making an unnecessary extra HTTP request for every single card.

File: trello_items.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        Card: The newly created Card object.
+    """
+    url = "https://api.trello.com/1/cards"
+    query = {
+        'key': key,
+        'token': token,
+        'name': title, 
+        'desc': description,
+        'pos': len(get_cards_for_board()) + 1, 
+        'idList': '5f3a9a92b421455eaa2ca175' # this is the id for the 'To Do' list
```
Comment (hugh-emerson) [inline trello_items.py]: You have hardcoded (and undocumented!) ids here, but elsewhere you do things like `if (get_list_name(card['id']) == 'To Do')`.

That's a bit inconsistent. You should make these clearly defined constants and you can write things like `if(card['id'] == TODO_LIST_ID)`.

Or you should update this line (and the other hardcoded ids below) by making a Trello request to find the correct list ID.

File: trello_items.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    url = "https://api.trello.com/1/boards/{}/cards".format(board_id)
+    cards = requests.request("GET", url, params=query)
+    for card in cards.json():
+        if (get_list_name(card['id']) == 'To Do'):
+            to_do_cards.append(Card(card['id'], card['pos'], card['name'], card['desc'], get_list_name(card['id'])))
+    return to_do_cards
+
+def get_cards_for_board():
+    """
+    Fetches all To Do cards from the board.
```
Comment (hugh-emerson) [inline trello_items.py]: This description is exactly the same as the previous function "get_todo_cards_for_board".
Did you forget to update it?

Also, there's duplicated code. 

But most importantly, do you want to just have this function and delete the other function? Currently only "To Do" items are displayed on your app - it's not possible to view done cards or move them back to "to do"

**Review by hugh-emerson — CHANGES_REQUESTED** [review]
Almost there. I've left a few suggestions, but the important one that you need to fix before I can approve this is the app currently only displays items with status == "To Do".

File: .gitignore (hunk 1 above)
Comment (MadhuRoy-git) [reply .gitignore]: Moved to .env file

File: requirements.txt (hunk 2 above)
Comment (MadhuRoy-git) [reply requirements.txt:4]: done

File: templates/index.html (hunk 3 above)
Comment (MadhuRoy-git) [reply templates/index.html]: Done

File: trello_items.py (hunk 4 above)
Comment (MadhuRoy-git) [reply trello_items.py]: Updated as I have a new version of Python

File: trello_items.py (hunk 5 above)
Comment (MadhuRoy-git) [reply trello_items.py]: Done

File: trello_items.py (hunk 6 above)
Comment (MadhuRoy-git) [reply trello_items.py]: Done

File: trello_items.py (hunk 7 above)
Comment (MadhuRoy-git) [reply trello_items.py]: Done

**Comment by hugh-emerson** [comment]
Looks good. Happy for you to merge this.

Only point I'd make is you should update the README to mention the necessary setup (creating a trello account, setting the correct variables in .env, setting the list ids in trello_items.py)

Also, you need to re-submit this in Aptem for it to be tracked correctly.

**Pull request merged**